### PR TITLE
chore: remove unneeded conversion webhooks + introduce unit tests for conversion functions

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR

- removes unneeded conversion webhooks from spike versions
- adapt conversion functions
- introduces unit tests for conversion functions
- removes conversion webhook tests from ginkgo testsuite
- bumps ginkgo version to v2
- introduce codecov.yaml for configuring test-coverage computations

### Related Issues

Part of #370

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -86,3 +85,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace golang.org/x/net => golang.org/x/net v0.7.0
```
[the remark this answers is not in the document]
Comment (odubajDT) [reply go.mod:89]: yes, there is a vulnerability in v0.5.0, therefore it was needed

File: .github/workflows/pr-checks.yml [hunk 2]
```diff
@@ -71,6 +71,41 @@ jobs:
         run: make test
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v3
+        with:
+          flags: tests
+
+  unit-test:
```
[the remark this answers is not in the document]
Comment (odubajDT) [reply .github/workflows/pr-checks.yml:77]: unit-tests is a separate job, but the test-coverage of the code is computed together

[the remark this answers is not in the document]
Comment (skyerus) [reply .github/workflows/pr-checks.yml:77]: Unless I'm mistaken, `make test` covers all tests whereas `make unit-test` covers a subset of the tests (under the apis dir). My concern is that if more unit tests are introduced in a different dir they won't be covered by this job. The pattern I've seen elsewhere is to run the unit tests with the test `-short` flag and check the flag in non-unit tests - https://github.com/open-feature/flagd/blob/main/flagd/tests/integration/caching_test.go#L20

[the remark this answers is not in the document]
Comment (odubajDT) [reply .github/workflows/pr-checks.yml:83]: adapted

File: Makefile [hunk 3]
```diff
@@ -67,7 +67,17 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
```
Comment (skyerus) [inline Makefile:69]: Is the `test` name still appropriate given it's now only targeting controllers/webhooks? 🤔 
I'm not sure what appropriate naming is best here but I think the nuances of `make test` vs `make unit-test` may cause confusion

Comment (odubajDT) [reply Makefile:69]: sure, we can change it, `unit-test` and `component-test` might be good?

Comment (skyerus) [reply Makefile:69]: That sounds good. Maybe `test` can run both?

Comment (odubajDT) [reply Makefile:69]: I would rather keep it separated, it's not that big of a deal to run 2 make commands

Comment (skyerus) [reply Makefile:69]: Fine by me

Comment (odubajDT) [reply Makefile:69]: adapted

**Review by Kavindu-Dodan — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -65,9 +65,19 @@ fmt: ## Run go fmt against code.
 vet: ## Run go vet against code.
 	go vet ./...
 
-.PHONY: test
+.PHONY: component-test
 test: manifests generate fmt vet envtest ## Run tests.
```
Comment (skyerus) [inline Makefile:69]:
```suggestion
component-test: manifests generate fmt vet envtest ## Run tests.
```

**Review by skyerus — APPROVED** [review]

**Pull request merged**